src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisClientLibConfiguration.java

```diff
@@ -1290,7 +1319,13 @@ public KinesisClientLibConfiguration withMaxCacheByteSize(final int maxCacheByte
     }
 
     public KinesisClientLibConfiguration withDataFetchingStrategy(String dataFetchingStrategy) {
-        recordsFetcherFactory.setDataFetchingStrategy(DataFetchingStrategy.valueOf(dataFetchingStrategy));
+        switch (dataFetchingStrategy.toUpperCase()) {
+            case "PREFETCH_CACHED":
+                recordsFetcherFactory.setDataFetchingStrategy(DataFetchingStrategy.PREFETCH_CACHED);
+                break;
+            default:
+                recordsFetcherFactory.setDataFetchingStrategy(DataFetchingStrategy.DEFAULT);
```

Comment:
Should this be a default, or throw an exception?

Reply:
Using value of takes care of this. It throws an IllegalArgumentException for unknown values.